client/request_handler_test.go

```diff
@@ -15,35 +15,41 @@ func NewRequestHandler(accountSid string, authToken string) *client.RequestHandl
 func TestRequestHandler_BuildUrlSetRegion(t *testing.T) {
 	// Region set via url
 	requestHandler := NewRequestHandler("user", "pass")
-	assert.Equal(t, "https://api.region.twilio.com", requestHandler.BuildUrl("https://api.region.twilio.com"))
+	assert.Equal(t, "https://api.region.twilio.com", assertAndGetURL(t, requestHandler, "https://api.region.twilio.com"))
 
 	// Region set via requestHandler
 	requestHandler.Region = "region"
-	assert.Equal(t, "https://api.region.twilio.com", requestHandler.BuildUrl("https://api.twilio.com"))
-	assert.Equal(t, "https://api.region.twilio.com", requestHandler.BuildUrl("https://api.urlRegion.twilio.com"))
+	assert.Equal(t, "https://api.region.twilio.com", assertAndGetURL(t, requestHandler, "https://api.twilio.com"))
+	assert.Equal(t, "https://api.region.twilio.com", assertAndGetURL(t, requestHandler, "https://api.urlRegion.twilio.com"))
 }
 
 func TestRequestHandler_BuildUrlSetEdgeDefaultRegion(t *testing.T) {
 	// Edge set via client
 	requestHandler := NewRequestHandler("user", "pass")
 	requestHandler.Edge = "edge"
-	assert.Equal(t, "https://api.edge.us1.twilio.com", requestHandler.BuildUrl("https://api.twilio.com"))
+	assert.Equal(t, "https://api.edge.us1.twilio.com", assertAndGetURL(t, requestHandler, "https://api.twilio.com"))
 }
 
 func TestRequestHandler_BuildUrlSetEdgeRegion(t *testing.T) {
 	//Edge and Region set via url
 	requestHandler := NewRequestHandler("user", "pass")
-	assert.Equal(t, "https://api.edge.region.twilio.com", requestHandler.BuildUrl("https://api.edge.region.twilio.com"))
+	assert.Equal(t, "https://api.edge.region.twilio.com", assertAndGetURL(t, requestHandler, "https://api.edge.region.twilio.com"))
 
 	// Edge and Region set via client
 	requestHandler.Edge = "edge"
-	assert.Equal(t, "https://api.edge.region.twilio.com", requestHandler.BuildUrl("https://api.region.twilio.com"))
+	assert.Equal(t, "https://api.edge.region.twilio.com", assertAndGetURL(t, requestHandler, "https://api.region.twilio.com"))
 	requestHandler.Region = "region"
-	assert.Equal(t, "https://api.edge.region.twilio.com", requestHandler.BuildUrl("https://api.twilio.com"))
-	assert.Equal(t, "https://api.edge.region.twilio.com", requestHandler.BuildUrl("https://api.urlEdge.urlRegion.twilio.com"))
+	assert.Equal(t, "https://api.edge.region.twilio.com", assertAndGetURL(t, requestHandler, "https://api.twilio.com"))
+	assert.Equal(t, "https://api.edge.region.twilio.com", assertAndGetURL(t, requestHandler, "https://api.urlEdge.urlRegion.twilio.com"))
 }
 
 func TestRequestHandler_BuildHostRawHostWithoutPeriods(t *testing.T) {
 	requestHandler := NewRequestHandler("user", "pass")
-	assert.Equal(t, "https://prism_twilio:4010", requestHandler.BuildUrl("https://prism_twilio:4010"))
+	assert.Equal(t, "https://prism_twilio:4010", assertAndGetURL(t, requestHandler, "https://prism_twilio:4010"))
+}
+
```

Comment:
Hi @kpritam! Thanks for your contribution! Could you add some tests to verify that errors are returned properly?

Reply:
@JenniferMah have added test, please have a look.